### ISSUE TITLE
enable_platform_analyzer_on_pre_net5_target does not appear to work

### ISSUE THREAD

**Issue by RobSwDev** [dotnet/roslyn-analyzers#5841]
I want to use the inbuilt analyzers on code targeting NetStandard2.0. I thought this was possible by setting `enable_platform_analyzer_on_pre_net5_target`  to `true` in the editorconfig.
I have the following code:
```
        public static void X()
        {
            var x = new System.Text.UTF7Encoding();
            var y = new System.Collections.ArrayList();
            Console.WriteLine(Console.WindowWidth);
            Console.Beep(1,2);
        }
```
I thought `UTF7Encoding`, `ArrayList`, `WindowWidth` (possibly) and `Console.Beep` should all produce warnings, but no warnings are shown when I build targeting NetStandard 2.0.
If I switch to target 5.0, I do see warnings for `UTF7Encoding` and `Console.Beep`.

How can I get these warnings when my project targets NetStandard2.0?

Is there an equivalent analyzer that covers everything that was caught by the now deprecated [Microsoft.DotNet.Analyzers.Compatibility](https://github.com/dotnet/platform-compat) package? E.g. use of `ArrayList` or `SecureString`? There are 9 `DE` and 4 `PC` rules that no longer seem to have an equivalent in the new world.

### Analyzer

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1416#configure-code-to-analyze`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

**Comment by RobSwDev** [comment]
[NewAnalyzers.zip](https://github.com/dotnet/roslyn-analyzers/files/8002850/NewAnalyzers.zip)
Attached a project showing this issue (for VS2022)

**Comment by buyaa-n** [comment]
The analyzer would work with that configuration though it is not warning on those APIs because they are not annotated with OSPlatformAttributes in netstandard, and there is no way to get warnings unless they annotated in netstandard as needed

Tagging @terrajobst @jeffhandley for their opinions/suggestions about your questions

**Comment by terrajobst** [comment]
@buyaa-n is correct. `Microsoft.DotNet.Analyzers.Compatibility` was a proof-of-concept, which is why it was in preview. It worked by carrying a database of APIs. We have changed this by adding attributes to the platform (see this [design for details](https://github.com/dotnet/designs/blob/main/accepted/2020/platform-exclusion/platform-exclusion.md)). We knew that this meant losing .NET Standard 2.0 support but felt that was necessary in order to deliver a much better experience moving forward, as carrying data files in the analyzer doesn't scale very well and also doesn't allow us to make these annotations part of the product design.

So I'm afraid this feature omission is by-design :-(
